docs/account.md

```diff
@@ -1,5 +1,219 @@
+
+#### Address
+
+```js
+type Address = {|
+  address: string,
+  derivationPath: string
+|};
+```
+
+#### BalanceHistoryMap
+
+TODO
+
+#### Operation
+
+TODO
+
+### TokenAccount
+
+```js
+{
+  type: "TokenAccount",
+  id: string,
+  parentId: string,
+  token: TokenCurrency,
+  balance: BigNumber,
+  operationsCount: number,
+  operations: Operation[],
+  pendingOperations: Operation[],
+  balanceHistory?: BalanceHistoryMap
+}
+```
+
+TODO EACH SINGLE FIELD TO BE DETAILED
+
+### ChildAccount
+
+```js
+{
+  type: "ChildAccount",
+  id: string,
+  name: string,
+  parentId: string,
+  currency: CryptoCurrency,
+  address: string,
+  balance: BigNumber,
+  operationsCount: number,
+  operations: Operation[],
+  pendingOperations: Operation[],
+  balanceHistory?: BalanceHistoryMap
+}
+```
+
+**Why does it differ from TokenAccount?**
+
+First of all, `TokenAccount` is associated to a `TokenCurrency` which allows a series of constraint, not to a `CryptoCurrency`. The key different is a TokenAccount will always be using the parent address to receive funds, it only exists in the context of the parent account and is highly associated to it. ChildAccount, on the contrary is more standalone, have its own address. (example of Tezos KT addresses)
+
+Apart from this important distinction, the two concept share many concepts and fields, that's why we have `SubAccount` an union type between the two.
+
+### union and utility types
+
+```js
+type SubAccount = TokenAccount | ChildAccount;
+
+type AccountLike = Account | SubAccount;
+
+type AccountLikeArray =
+  | AccountLike[]
+  | TokenAccount[]
+  | ChildAccount[]
+  | Account[];
+```
+
+> NB: `AccountLikeArray` is here because `AccountLike[]` does not practically works because FlowType limitation.
+
+## "Raw types"
+
+In order to allow serialization of the types displayed above we need to have intermediary conversion types (suffixed by `*Raw`) that we can directly JSON.stringify and save to a database and symetrically load up back into the original types.
+
+Therefore, we introduce these types:
+
+```js
+type TokenAccountRaw = ...;
+type ChildAccountRaw = ...
+type AccountRaw = ...;
+type OperationRaw = ...;
+// as well as union types
+type SubAccountRaw = TokenAccountRaw | ChildAccountRaw;
+type AccountRawLike = AccountRaw | SubAccountRaw;
+```
+
+Typically, BigNumber will be serialized to a decimal string.
+
+This indirection also allows us the flexibility to ease backwards compatibility: **we must guarantee that existing raw objects from 6 months ago are stilled supported** to avoid having complex migration system. The convention is to never changes a field name and each time we introduce fields, we make them optional (`field?: T`) in order to support the absence of it, which typically would be "defaulted" during the conversion back to the main types allowing us to forget about this problem and sometimes do breaking change in the main model (in respect of not breaking the raw version).
```

Comment:
```suggestion
This indirection also allows us the flexibility to ease backward compatibility: **we must guarantee that existing raw objects from 6 months ago are stilled supported** to avoid having a complex migration system. The convention is to never change a field name and each time we introduce fields, we make them optional (`field?: T`) in order to support the absence of it, which typically would be "defaulted" during the conversion back to the main types allowing us to forget about this problem and sometimes do breaking change in the main model (in respect of not breaking the raw version).
```